flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/SessionContext.java

```diff
@@ -40,6 +44,9 @@
 
 	private final Map<String, ViewEntry> views;
 
+	private Optional<String> currentCatalog = Optional.empty();
```

Comment:
Could we just put the current catalog and into the `sessionProperties`? Then handling of merging environment files would happen automatically and we could reduce the amount of changes in this PR.